### PR TITLE
XML Parsing Cleanup

### PR DESCRIPTION
**Pull request by barronpm**
**Changes**
- De-duplicates some code
- Adds extension methods for common operations
- Cleans up some formatting

Supercedes #5992

### CODE REVIEW

File: MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        if (DateTime.TryParse(
+                text,
+                CultureInfo.InvariantCulture,
+                DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal,
+                out value))
+        {
+            return true;
+        }
+
+        logger.LogWarning("Invalid date: {Date}", text);
```
Comment (cvium) [inline MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs:64]: Feels a bit weird to put a logger in the args

Comment (barronpm) [reply MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs:64]: I agree, I just added it to keep existing behavior. If you're fine with not logging on failure anymore I'll remove it.

File: MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+                    break;
+                case "type":
+                case "Type":
+                    Enum.TryParse(subtree.ReadElementContentAsString(), true, out type);
+                    break;
+                case "order":
+                case "sortorder":
+                case "SortOrder":
+                    if (int.TryParse(
```
Comment (cvium) [inline MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs:143]: shouldn't it be using the TryReadInt here?

File: MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    /// <param name="reader">The <see cref="XmlReader"/>.</param>
+    /// <returns>IEnumerable{System.String}.</returns>
+    public static IEnumerable<string> GetStringArray(this XmlReader reader)
+    {
+        ArgumentNullException.ThrowIfNull(reader);
+        var value = reader.ReadElementContentAsString();
+
+        // Only split by comma if there is no pipe in the string
+        // We have to be careful to not split names like Matthew, Jr.
+        var separator = !value.Contains('|', StringComparison.Ordinal)
```
Comment (cvium) [inline MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs:180]: The rest of this method seems a bit inefficient

File: MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs (hunk 2 above)
Comment (barronpm) [reply MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs]: Yes, I've updated this method to use the other extension methods

File: MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs (hunk 1 above)
Comment (cvium) [reply MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs:64]: I don't think we've gotten that many reports about this, so it seems a little useless

File: MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        }
+
+        if (string.IsNullOrWhiteSpace(name))
+        {
+            return null;
+        }
+
+        return new PersonInfo
+        {
+            Name = name.Trim(),
```
Comment (cvium) [inline MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs:160]: It's already trimmed

File: MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs (hunk 3 above)
Comment (barronpm) [reply MediaBrowser.Controller/Extensions/XmlReaderExtensions.cs:170]: Yeah, this is an ideal case for spans, but we don't have a `Split` method for multiple needles.
One solution is to do a pure fallback of `|` -> `;` -> `,` for separators, which is technically a breaking change, but I doubt there's a ton of files out there using both.

Alternatively, we would have to make another extension method, but imo that's a big enough change to warrant a separate PR.

**Review by cvium — APPROVED** [review]

**Review by Shadowghost — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**